registry/storage/driver/s3/s3.go

```diff
@@ -597,11 +597,11 @@ func (d *driver) List(path string) ([]string, error) {
 
 	for {
 		for _, key := range listResponse.Contents {
-			files = append(files, strings.Replace(key.Key, d.s3Path(""), "", 1))
+			files = append(files, "/"+strings.Replace(key.Key, d.s3Path(""), "", 1))
```

Comment:
This should probably use `path.Join`. The result will be the same but it is more likely to handle odd edge cases. I'm unfamiliar with the path management in the s3 driver, so please double check this is correct.